x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -290,17 +315,20 @@ public CacheFile getCacheFile(CacheKey cacheKey, long fileLength) throws Excepti
 
     @Override
     public IndexInput openInput(final String name, final IOContext context) throws IOException {
-        ensureOpen();
+        return openInput(fileInfo(name), context);
+    }
 
-        final BlobStoreIndexShardSnapshot.FileInfo fileInfo = fileInfo(name);
+    private IndexInput openInput(final BlobStoreIndexShardSnapshot.FileInfo fileInfo, final IOContext context) {
```

Comment:
Splitting this method into two allows to open an `IndexInput` even if the snapshot is not marked as `loaded` yet

Reply:
Indeed - I pushed c8a1c6bf7e480d4961fe1e4a6d7fb8ab01e85534 to remove the method.